File: API/RequestHelpers/ProductParams.cs
namespace API.RequestHelpers;

/// <summary>
/// Represents filtering, searching, and sorting parameters
/// for querying products, including pagination support.
/// </summary>
public class ProductParams : PaginationParams
{
    /// <summary>
    /// The property by which to order products.
    /// Possible values could include "price", "priceDesc", or "name".
    /// </summary>
    public string? OrderBy { get; set; }

    /// <summary>
    /// The search keyword used to match product names.
    /// This is typically a partial or full product name entered by the user.
    /// </summary>
    public string? SearchTerm { get; set; }

    /// <summary>
    /// A comma-separated list of brand names used to filter products.
    /// Example: "nike,adidas,puma".
    /// </summary>
    public string? Brands { get; set; }

    /// <summary>
    /// A comma-separated list of product types used to filter results.
    /// Example: "shoes,shirts,accessories".
    /// </summary>
    public string? Types { get; set; }
}

File: client/src/app/models/productParams.ts
// Type representing query parameters for fetching/filtering products
export type ProductParams = {
    orderBy: string;      // Field or criteria to order/sort products (e.g., "price", "name")
    searchTerm?: string;  // Optional search term to filter products by name or description
    types: string[];      // Array of product types/categories to filter by
    brands: string[];     // Array of product brands to filter by
    pageNumber: number;   // Current page number for pagination
    pageSize: number;     // Number of products per page
}
